terraform/aws-public/mesos-slaves.tf

```diff
@@ -8,7 +8,7 @@ resource "atlas_artifact" "mesos-slave" {
 /* Mesos slave instances */
```

Comment:
Dont think that interpolation is supported by terraform. ->  "${atlas_artifact.mesos-master.metadata_full.region-${var.amis.${region}}}"

Think we can harcode it as ${atlas_artifact.mesos-master.metadata_full.region-eu-west-1}
or create a static map region -> ami-id
Then just ami = lookup(var.amis, var.region)

Static mapping would need to be updated every time we build images and the atlas_artifact resource shoudl be removed as it would be useless if we go for second option.